tests/testthat/test-parallel-with-snowfall.R

```diff
@@ -0,0 +1,70 @@
+# Ensure the latest precompiled version of FIMS is installed in R before
+# running devtools. To do this, either run:
+# - devtools::install() followed by devtools::test(), or
+# - devtools::check()
+
+# Run FIMS in serial and parallel
+# This test demonstrates how to run the FIMS model in both serial and parallel
+# modes. The test compares the execution time and results of running the model
+# in serial versus parallel. The parallel execution uses the {snowfall} package
+# to parallelize the tasks across multiple CPU cores
+
+# Load the model comparison operating model data from the fixtures folder
+load(test_path("fixtures", "integration_test_data.RData"))
+
+# Run the FIMS model in serial and record the execution time
+estimation_results_serial <- vector(mode = "list", length = length(om_input_list))
+
+start_time_serial <- Sys.time()
+for (i in 1:length(om_input_list)) {
+  estimation_results_serial[[i]] <- setup_and_run_FIMS(
+    iter_id = i,
+    om_input_list = om_input_list,
+    om_output_list = om_output_list,
+    em_input_list = em_input_list,
+    estimation_mode = TRUE)
+}
+end_time_serial <- Sys.time()
+estimation_time_serial <- end_time_serial - start_time_serial
+
+test_that("Run FIMS in parallel using {snowfall}", {
+
+  core_num <- parallel::detectCores() - 1
+  snowfall::sfInit(parallel = TRUE, cpus = core_num)
+  snowfall::sfLibrary(FIMS)
```

Comment:
I'm not sure why it ran without loading FIMS in the child processes. Had to add it to run locally, feel free to remove if you want.

Reply:
According to the guidelines from Advanced R, we should avoid including `library()` calls in the test files. Could you try running `devtools::install()` first and then execute the tests (e.g., devtools::test()) without using library(FIMS) to see if it works locally? Thank you!